libr/core/cmd_magic.c

```diff
@@ -79,12 +71,12 @@ static int r_core_magic_at(RCore *core, const char *file, ut64 addr, int depth,
 	}
 //repeat:
 	//if (v) r_cons_printf ("  %d # pm %s @ 0x%"PFMT64x"\n", depth, file? file: "", addr);
-	if (delta+2>core->blocksize) {
+	if (2>core->blocksize) {
```

Comment:
if you pass a block, then pass the length, otherwise its inconsistent, and im not sure if we really want to mess the apis more with this because we will probably need to use RBuffer or use a struc thtat contains addr+buffer+length instead of passing 3 parameters all the time